src/s1ap/handlers/s1ap_msg_delegator.c

```diff
@@ -2438,3 +2442,170 @@ int convertUeHoCancelToProtoIe(InitiatingMessage_t *msg,
 
     return 0;
 }
+
+int convertErabModIndToProtoIe(InitiatingMessage_t *msg, struct proto_IE *proto_ies)
+{
+    proto_ies->procedureCode = msg->procedureCode;
+    proto_ies->criticality = msg->criticality;
+    int no_of_IEs = 0;
+
+    if (msg->value.present
+            == InitiatingMessage__value_PR_E_RABModificationIndication)
+    {
+        ProtocolIE_Container_129P77_t *protocolIes =
+                &msg->value.choice.E_RABModificationIndication.protocolIEs;
+        no_of_IEs = protocolIes->list.count;
+        proto_ies->no_of_IEs = no_of_IEs;
+
+        log_msg(LOG_INFO, "No of IEs = %d\n", no_of_IEs);
+        proto_ies->data = calloc(sizeof(struct proto_IE_data), no_of_IEs);
+        
+        for (int i = 0; i < protocolIes->list.count; i++)
+        {
+            E_RABModificationIndicationIEs_t *ie_p;
+            ie_p = protocolIes->list.array[i];
+            switch (ie_p->id)
+            {
+            case ProtocolIE_ID_id_eNB_UE_S1AP_ID:
+            {
+                ENB_UE_S1AP_ID_t *s1apENBUES1APID_p = NULL;
+                if (E_RABModificationIndicationIEs__value_PR_ENB_UE_S1AP_ID
+                        == ie_p->value.present)
+                {
+                    s1apENBUES1APID_p = &ie_p->value.choice.ENB_UE_S1AP_ID;
+                }
+
+                if (s1apENBUES1APID_p == NULL)
```

Comment:
else candidate. Same comment for other places as well.

Reply:
Done